R/distribution_formulas.R

```diff
@@ -0,0 +1,370 @@
+#' Validaity checks for new_data_distribution and new_process_distribution
+#' 
+check_distribution_validity <- function(args){
+  family <- args$family
+  families <- args$families
+  sd <- args$sd
+  if (class(family) !=  "family") {
+    cli::cli_abort(c(
+      "{.code {family}} is the incorrect type",
+      "*" = "family needs to be specified as a family class",
+      "*" = " e.g. `family = gaussian()`."))
+  }
+  if (!(family[["family"]] %in% families)) {
+    fam_name <- family[["family"]]
+    cli::cli_abort(c(
+      "FIMS currently does not offer the family, {.code {fam_name}}. for this distribution type",
+      "*" = "The families available for this distribution type are:",
+      "*" = "{.code {families}}."))
+
+  }
+  if(!is.null(args$data_type)){
+    data_type <- args$data_type
+    data_type_names <- c("index", "cpue", "agecomp", "lengthcomp")
+
+    if (!(data_type %in% data_type_names)) {
+      cli::cli_abort(c(
+        "The data type is incorrect.",
+        "*" = "Correct values are:",
+        "*" = "{.code {data_type_names}}."
+      ))
+    }
+
+    if ((data_type == "agecomp" || data_type == "lengthcomp") &&
```

Comment:
I think we can change this to `if (grepl("comp", data_type)) &&` to be more generic.